src/getViewForSwagger2.ts

```diff
@@ -1,51 +1,111 @@
-'use strict';
+// TODO: This file needs some love <3
+import {
+    transform,
+    forEach,
+    camelCase,
+    merge,
+    isArray,
+    isString
+} from 'lodash';
+import { convertType } from './typescript';
+import { Security } from './swagger/Swagger';
+import { CodeGenOptions } from './options/options';
+import { TypeSpec } from './typespec';
+
+export type GenerationTargetType = 'typescript' | 'custom';
+
+interface Header {}
+
+interface Method {
+    methodName: string;
+    intVersion: number;
+    isLatestVersion: boolean;
+    isSecure: boolean;
+    isSecureToken: boolean;
+    isSecureApiKey: boolean;
+    isSecureBasic: boolean;
+    path: string;
+    pathFormatString: string;
+    className: string;
+    version: string;
+    method: string;
+    isGET: boolean;
+    isPOST: boolean;
+    summary: string;
+    externalDocs: string;
+    parameters: string[];
+    headers: Header[];
+    successfulResponseType: string;
+    successfulResponseTypeIsRef: boolean;
+}
+
+interface Definition {
+    name: string; 
+    description: string | undefined;
+    tsType: TypeSpec
+}
+
+export interface ViewData {
+    isES6: boolean;
+    description: string;
+    isSecure: boolean;
+    moduleName: string;
+    className: string;
+    imports: ReadonlyArray<string>;
+    domain: string;
+    isSecureToken: boolean;
+    isSecureApiKey: boolean;
+    isSecureBasic: boolean;
+    methods: Method[];
+    definitions: Definition[];
+}
 
-var fs = require('fs');
-var Mustache = require('mustache');
-var beautify = require('js-beautify').js_beautify;
-var lint = require('jshint').JSHINT;
-var _ = require('lodash');
-var ts = require('./typescript');
+interface LatestMethodVersion {
+    [index: string]: number;
+}
 
-var defaultSuccessfulResponseType = 'void';
+const defaultSuccessfulResponseType = 'void';
 
-var normalizeName = function(id) {
+function normalizeName(id: string): string {
     return id.replace(/\.|\-|\{|\}/g, '_');
 };
 
-var getPathToMethodName = function(opts, m, path){
+function getPathToMethodName(__: CodeGenOptions, m: string, path: string): string {
```

Comment:
does this method need the first argument?

Reply:
Nope it's unused, but this whole file needs a bit of love, it's the one part I didn't get time to spent a lot of time on.

I left most of the original code in tact here and just made some stylistic changes and added the typings.